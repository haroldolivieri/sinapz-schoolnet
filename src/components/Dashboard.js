import React from 'react';
import Grid from "@material-ui/core/Grid";
import { DashboardCard, DashboardPaper } from '.'
import PropTypes from 'prop-types'

export const DashboardIndice = ({
    quantidade = 0,
    tipo = 'Unknown',
}) => (
    <Grid item md={4}>
        <DashboardCard
            // backgroundColor="#00f"
            quantidade = { quantidade }
            tipo = { tipo }
        />
    </Grid>
)
DashboardIndice.propTypes = {
    quantidade: PropTypes.number,
    tipo: PropTypes.string,
}

export const DashboardAvaliacao = ({
    tipo = 'Unknown',
    disciplina = 'Unknown',
    grupo = 'Unknown',
    tempoRestante = 'Unknown'
}) => (
    <DashboardPaper
        tipo = { tipo }
        disciplina = { disciplina }
        grupo = { grupo}
        tempoRestante = { tempoRestante }
    />
)
DashboardAvaliacao.propTypes = {
    tipo: PropTypes.string,
    disciplina: PropTypes.string,
    grupo: PropTypes.string,
    tempoRestante: PropTypes.string,
}

const Dashboard = () => (
    <div>
        <div>
            <Grid container spacing={24}>
            {/* Start DashboardIndices Temp */}
                <Grid item md={4}>
                    <DashboardCard
                        // backgroundColor="#00f"
                        quantidade = "5"
                        tipo = "Filiais"
                    />
                </Grid>
                <Grid item md={4}>
                    <DashboardCard
                        // backgroundColor="#00f"
                        quantidade = "40"
                        tipo = "Professores"
                    />
                </Grid>
                <Grid item md={4}>
                    <DashboardCard
                        // backgroundColor="#00f"
                        quantidade = "200"
                        tipo = "Alunos"
                    />
                </Grid>
            {/* End Temp */}
            {/* {DashboardIndices.map(item => (
                    <DashboardIndice
                        quantidade={item.quantidade}
                        tipo={item.tipo}
                    />
                ))} */}
            </Grid>
        </div>
        <div>
            <h1>Próximas Avaliações em 24h </h1>
            {/* Start DashboardAvaliacoes Temp */}
            <DashboardPaper
                tipo={"Prova"}
                disciplina={ "Geografia" }
                grupo = { "Ensino Médio"}
                tempoRestante={ "30m" }
            />
            <br/>
            <DashboardPaper
                tipo={"Teste"}
                disciplina={ "Matematica" }
                grupo = { "Ensino Fundamental"}
                tempoRestante={ "1h" }
            />
            <br/>
            <DashboardPaper
                tipo={"Prova"}
                disciplina={ "Geografia" }
                grupo = { "Ensino Médio"}
                tempoRestante={ "2h" }
            />
            <br/>
            <DashboardPaper
                tipo={"Prova"}
                disciplina={ "Geografia" }
                grupo = { "Ensino Médio"}
                tempoRestante={ "30m" }
            />
            <br/>
            <DashboardPaper
                tipo={"Teste"}
                disciplina={ "Matematica" }
                grupo = { "Ensino Fundamental"}
                tempoRestante={ "1h" }
            />
            <br/>
            <DashboardPaper
                tipo={"Prova"}
                disciplina={ "Geografia" }
                grupo = { "Ensino Médio"}
                tempoRestante={ "2h" }
            />
            <br/>
            {/* End DashboardAvaliacoes Temp */}
            {/* {DashboardAvaliacoes.map(item => (
                    <DashboardAvaliacao
                        tipo={item.tipo}
                        disciplina={item.disciplina}
                        grupo={item.grupo}
                        tempoRestante={item.tempoRestante}
                    />
                ))} */}
        </div>
    </div>
)

export default Dashboard
