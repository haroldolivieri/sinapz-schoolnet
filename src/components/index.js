export { default as NotFound } from './NotFound'
export * from './NotImplementedYet'
export { default as ResponsiveDrawer } from './ResponsiveDrawer'
export { default as SideDrawer } from './SideDrawer'
export { default as Dashboard } from '../containers/Dashboard'
export { default as DashboardCard } from './DashboardCard'
export { default as DashboardPaper } from './DashboardPaper'
