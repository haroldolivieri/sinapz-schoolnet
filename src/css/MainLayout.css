.app-frame {
    display: flex;
    width: 100%;
}

.toolbar {
    display: flex;
    flex-direction: row;
    height: 8em;
    flex-wrap: wrap;
    background-color: white;
    position: fixed;
    justify-content: space-between;
    align-items: center;
    right: 0;
    left: 0;
    padding: 2em;
}
@media only screen and (min-width: 900px) {
    .toolbar {
        left: 30vw;
        height: 6em;
    }
}
@media only screen and (min-width: 1200px) {
    .toolbar {
        left: 20vw;
    }
}

.logo {
    width: 15em;
    left: -1em;
    position: relative;
    margin: 0 auto;
    align-self: center;
}
@media only screen and (min-width: 900px) {
    .logo {
        left: 0;
        margin: 0;
    }
}

.search {
    margin-top: 1em;
    width: 100vw;
}
@media only screen and (min-width: 900px) {
    .search {
        margin-top: 0;
        width: 30vw;
    }
}

.children-content {
    width: 100vw;
    padding: 2em;
    background-color: '#FAF8F8';
    min-height: 80vh;
    margin-top: 10em;
}
@media only screen and (min-width: 900px) {
    .children-content {
        width: 70vw;
        margin-left: 30vw;
        margin-top: 8em;
    }
}
@media only screen and (min-width: 1200px) {
    .children-content {
        margin-left: 20vw;
        width: 80vw;
    }
}