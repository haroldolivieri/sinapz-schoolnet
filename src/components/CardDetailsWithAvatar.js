import React from 'react'
import Card from '@material-ui/core/Card'
import CardHeader from '@material-ui/core/CardHeader'
import Avatar from '@material-ui/core/Avatar'
import '../css/CardDetailsWithAvatar.css'
import PropTypes from 'prop-types'
import { extractInitials } from '../util/handlerWithNames'

export const CardDetailsWithAvatar = ({branch}) => {
  return (
    <Card key={branch.id} className="card-container">
      <div className="avatar-item">
        <CardHeader avatar={
          <Avatar>{ extractInitials(branch.name) }</Avatar>
        }
        />
      </div>
      <div className="content-item">
        <span className="item-card-text-name">{branch.name}</span>
        <span className="item-card-text-sub">{`Respons√°vel: ${branch.inCharge}`}</span>
      </div>
    </Card>
  )
}

CardDetailsWithAvatar.propTypes = {
  branch: PropTypes.shape({
    id: PropTypes.string,
    name: PropTypes.string,
    inCharge: PropTypes.string
  })
}

CardDetailsWithAvatar.defaultProps = {
  branch: {
    id: "Unknown",
    name: "Unknown",
    inCharge: "Unknown"
  }
}

export default CardDetailsWithAvatar