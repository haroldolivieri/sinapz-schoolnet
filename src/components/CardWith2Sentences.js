import React from 'react';
import PropTypes from 'prop-types';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import '../css/CardWith2Sentences.css'

export const CardWith2Sentences = ({
                            FirstSentence = [<b className='highlight'>Hello</b>],
                            SecondSentence = [<b>]World!</b>],
}) => (
        <Card class="card">
            <CardContent  class="card_content">
                { FirstSentence }
                { SecondSentence }
            </CardContent>
        </Card>
);

CardWith2Sentences.propTypes = {
    FirstSentence: PropTypes.string,
    SecondSentence: PropTypes.string,
};