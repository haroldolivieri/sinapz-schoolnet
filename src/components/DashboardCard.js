import React from 'react';
import PropTypes from 'prop-types';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';
import '../css/DashboardCard.css'


function SimpleCard(props) {

    const {
        quantidade,
        tipo,
    } = props;

    return (
        <div class="card" >
            <Card>
                <CardContent  class="card_content">
                    <Typography variant="h1" component="h1" >
                        <b>{ quantidade }</b>
                    </Typography>
                    <Typography variant="h2" component="h2">
                        <b>{ tipo }</b>
                    </Typography>
                </CardContent>
            </Card>
        </div>
    );
}

SimpleCard.propTypes = {
    classes: PropTypes.object.isRequired,
};

export default 
    SimpleCard;
