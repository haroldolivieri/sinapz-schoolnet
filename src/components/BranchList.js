import React from 'react'
import CardDetailsWithAvatar from './CardDetailsWithAvatar'

const BranchList = ({ branches }) => {
  if(!branches || !branches.length){
    return <p>Essa rede n√£o tem filiais</p>
  }
  return branches.map(branch => {
    return(
      <CardDetailsWithAvatar branch={branch} key={branch.id} />
    )
  })
}

export default BranchList
