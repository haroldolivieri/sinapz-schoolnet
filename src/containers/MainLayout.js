import React, { Component } from 'react'
import MenuIcon from '@material-ui/icons/Menu'
import { ResponsiveDrawer, SideDrawer } from '../components'
import { loadInitalData } from '../menu/actions'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import SearchInput from '../containers/SearchInput'
import '../css/MainLayout.css'

export class MainLayout extends Component {
    constructor(props) {
        super(props)

        this.props.loadInitalData()
    }

    state = {
        navDrawerOpened: false,
        selectedItem: 'dashboard'
    }

    toggleDrawer = () => {
        this.setState(state => ({ navDrawerOpened: !state.navDrawerOpened }))
    }

    renderToolbar() {
        return (
            <div className="toolbar">
                <MenuIcon
                    className="only-mobile"
                    onClick={this.toggleDrawer}
                    aria-label="Open drawer"
                />

                <img
                    className="logo"
                    alt="logo"
                    src={require(`../images/logo.png`)}
                />

                <div className="search">
                    <SearchInput />
                </div>
            </div>
        )
    }

    render() {
        return (
            <div className="app-frame">
                <div className="only-mobile">
                    <ResponsiveDrawer
                        opened={this.state.navDrawerOpened}
                        onClose={this.toggleDrawer}
                    />
                </div>

                <div className="hidden-for-mobile">
                    <SideDrawer />
                </div>

                {/* Toolbar with hamburguer menu, logo and search input */}
                {this.renderToolbar()}

                {/* Content injected by router */}
                <div className="children-content">{this.props.children}</div>
            </div>
        )
    }
}

const mapDispatchToProps = dispatch =>
    bindActionCreators({ loadInitalData }, dispatch)
export default connect(
    null,
    mapDispatchToProps
)(MainLayout)
