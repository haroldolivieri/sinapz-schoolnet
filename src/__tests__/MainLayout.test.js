import React from 'react'
import MenuIcon from '@material-ui/icons/Menu'
import { MainLayout } from '../containers/MainLayout'
import { shallow } from 'enzyme'
import { ResponsiveDrawer } from '../components'

let wrapper = shallow(<MainLayout loadInitalData={function() {}} />)

describe('MainLayout', () => {
    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })

    describe('ResponsiveDrawer', () => {
        let responsiveDrawer
        beforeEach(() => (responsiveDrawer = wrapper.find(ResponsiveDrawer)))

        it('starts closed', () => {
            responsiveDrawer = wrapper.find(ResponsiveDrawer)
            expect(responsiveDrawer.props().opened).toBe(false)
        })
    })

    describe('Toolbar', () => {
        let toolbar
        beforeEach(() => (toolbar = wrapper.find('.toolbar')))

        it('hamburguer has .only-mobile class', () => {
            expect(toolbar.find(MenuIcon).hasClass('only-mobile')).toBeTruthy()
        })

        it('has logo', () => {
            expect(toolbar.find('.logo').exists()).toBeTruthy()
        })

        it('has search input', () => {
            expect(toolbar.find('.search').exists()).toBeTruthy()
        })
    })
})
