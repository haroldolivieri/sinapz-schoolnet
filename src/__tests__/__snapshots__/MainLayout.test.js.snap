// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayout renders correctly 1`] = `
<div
  className="app-frame"
>
  <div
    className="only-mobile"
  >
    <ResponsiveDrawer
      onClose={[Function]}
      opened={false}
    />
  </div>
  <div
    className="hidden-for-mobile"
  >
    <withRouter(Connect(SideDrawer)) />
  </div>
  <main
    className="main-content"
  >
    <div
      className="toolbar"
    >
      <div
        className="only-mobile"
      >
        <WithStyles(IconButton)
          aria-label="Open drawer"
          onClick={[Function]}
        >
          <pure(MenuIcon) />
        </WithStyles(IconButton)>
      </div>
    </div>
    <div />
  </main>
</div>
`;
